### PR TITLE
#22989 use defaults when not providing start or end date

### LINKED ISSUE #22989: Experiment Scheduling
### User Story
As a user, I need to set up the start and end dates to schedule and run an experiment.

### Acceptance Criteria

- The user can select a start date and time and an end date and time for the experiment.
- The dates and times selected are saved and displayed correctly.
- The time needs to be in 30-minute steps
- Allow clear dates

### dotCMS Version
master

### Proposed Objective
Core Features

### Proposed Priority
Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
_to_define_

### Assumptions & Initiation Needs
_to_define_

### Quality Assurance Notes & Workarounds
_to_define_

### Designs
Screenshot 2023-04-17 at 5 26 24 PM

Screenshot 2023-04-17 at 5 26 43 PM
Screenshot 2023-04-17 at 5 26 30 PM

### PR DESCRIPTION
**Pull request by dsilvam**
Does the following when setting the Scheduling of an Experiment:
* when only a start date is provided, the end date gets set as start date + 35 days (max Experiment length)
* when only the end date is provided, the start date is set as the end date - 35 days (max Experiment length) 

Postman tests included.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java [hunk 1]
```diff
@@ -174,14 +174,25 @@ public Experiment save(final Experiment experiment, final User user) throws
...
         if(experiment.targetingConditions().isPresent()) {
             saveTargetingConditions(experiment, user);
         }
 
-        final Experiment experimentToSave = builder.build();
+        Experiment experimentToSave = builder.build();
+
+        Optional<Experiment> existingExperiment = find(experimentToSave.id().get(), user);
+
+        if(experimentToSave.status() != RUNNING &&
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:187]: I know this code come from other PR.
Maybe we should extract this to a function returning a boolean for the first 3 evaluations:
```
static List<STATUS> SOME_NAME = List.of(RUNNING, ENDED, ARCHIVED);
...
boolean hasNotStarted(List<STATUS> statuses, Experiment experiment) {
  return !SOME_NAME.contains(experiment.status())
}
```

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java [hunk 2]
```diff
@@ -174,14 +174,25 @@ public Experiment save(final Experiment experiment, final User user) throws
...
+                experimentToSave.status() != ENDED &&
+                experimentToSave.status() != ARCHIVED &&
+                experimentToSave.scheduling().isPresent() &&
+                (existingExperiment.isPresent() && !existingExperiment.get().scheduling().
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:191]: Could be simplified as
```
existingExperiment
  .map(exp -> exp.scheduling)
  .map(sche -> sche.equals(experimentToSave.scheduling()))
  .orElse(false)
```

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java [hunk 3]
```diff
@@ -1067,13 +1071,11 @@ public Scheduling validateScheduling(final Scheduling scheduling) {
...
             DotPreconditions.checkState(scheduling.endDate().get().isAfter(NOW),
                     "Invalid Scheduling. End date is in the past");
-            DotPreconditions.checkState(
-                    Instant.now().plus(EXPERIMENTS_MAX_DURATION.get(), ChronoUnit.DAYS)
-                            .isAfter(scheduling.endDate().get()),
-                    "Experiment duration must be less than "
-                            + EXPERIMENTS_MAX_DURATION.get() +" days. ");
 
-            toReturn = scheduling.withStartDate(Instant.now());
+            final Instant startDate = scheduling.endDate().get().minus(EXPERIMENTS_MAX_DURATION.get(),
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:1075]: What if `scheduling.endDate()` is empty?

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:1075]: it is the case above this one

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java:191]: I don't see that being simpler

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java]: Simplified the condition and extracted second part to a private method

**Review by victoralfaro-dotcms — APPROVED** [review]

**Pull request merged**